### PR TITLE
fix(code quality): Remove panics from functions that return result

### LINKED ISSUE #951: code quality - remove panics from functions that return `Result`
Done
* Halo2::ipa::read_fr (as well as the FFI variant in the operator) ✅ 
* halo2::kzg::read_fr (as well as the FFI variant in the operator) ✅ 
* Batcher::listen_new_blocks ✅ 
* Batcher::handle_message ✅ 
* Batcher::finalize_batch ✅ 
* Batcher::submit_batch ✅ 
* Batcher::handle_nonpaying_msg ✅ 

Remaining, but dont really make anything break, just code quality
* Batcher::listen_connections => Remove 'expect'
* aligned_batcher::main
* aligned::main

### PR DESCRIPTION
**Pull request by PatStiles**
closes #951 

Removes possible panics in `aligned_batcher` and `aligned::main`.

### CODE REVIEW

File: batcher/aligned-batcher/src/main.rs [hunk 1]
```diff
@@ -46,7 +46,7 @@ async fn main() -> Result<(), BatcherError> {
     // spawn task to listening for incoming blocks
     tokio::spawn({
         let app = batcher.clone();
-        async move { app.listen_new_blocks().await.unwrap() }
+        async move { app.listen_new_blocks().await }
```
Comment (Oppen) [inline batcher/aligned-batcher/src/main.rs]: This seems to ignore the errors, is that correct?

File: batcher/aligned/src/main.rs [hunk 2]
```diff
@@ -307,7 +307,16 @@ async fn main() -> Result<(), AlignedError> {
...
                 warn!("Missing keystore used for payment. This proof will not be included if sent to Eth Mainnet");
-                LocalWallet::from_str(ANVIL_PRIVATE_KEY).expect("Failed to create wallet")
+                match LocalWallet::from_str(ANVIL_PRIVATE_KEY) {
+                    Ok(wallet) => wallet,
+                    Err(e) => {
+                        warn!(
+                            "Failed to create wallet from anvil private key: {}",
+                            e.to_string()
+                        );
+                        return Ok(());
```
Comment (Oppen) [inline batcher/aligned/src/main.rs:317]: Should the error be forwarded?

Comment (MauroToscano) [reply batcher/aligned/src/main.rs:317]: This is already the return of the main function. Not much to do, but we could create an issue to return an Error everywhere so the application ends with an Error

File: batcher/aligned-batcher/src/main.rs (hunk 1 above)
Comment (MauroToscano) [reply batcher/aligned-batcher/src/main.rs]: This change seems strange, the unwrap should stay, app can't recover from this

Comment (MauroToscano) [reply batcher/aligned-batcher/src/main.rs]: We can change it to an expect to make it more clear that we cannot recover from this

Comment (PatStiles) [reply batcher/aligned-batcher/src/main.rs]: `expect()` added

**Review by MauroToscano — APPROVED** [review]

File: batcher/aligned-batcher/src/types/errors.rs [hunk 3]
```diff
@@ -4,6 +4,7 @@ use ethers::types::SignatureError;
 use tokio_tungstenite::tungstenite;
 
 pub enum BatcherError {
+    TcpListenderError(String),
```
Comment (entropidelic) [inline batcher/aligned-batcher/src/types/errors.rs]:
```suggestion
    TcpListenerError(String),
```

**Review by uri-99 — APPROVED** [review]

**Pull request merged**